stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -242,6 +242,27 @@ public class ChatClient internal constructor(
         return createInitListenerCall { initListener -> setUser(user, tokenProvider, initListener) }
     }
 
+    /**
+     * Initializes [ChatClient] for a specific user and a given [userToken].
+     * Caution: This method doesn't establish connection to the web socket, you should use [connectUser] instead.
+     *
+     * This method initializes [ChatClient] to allow the use of Stream REST API client.
+     * Moreover, it warms up the connection, and sets up notifications.
+     *
+     * @param user the user to set
+     * @param userToken the user token
+     */
+    public fun setUserWithoutConnecting(user: User, userToken: String) {
+        if (isUserSet()) {
```

Comment:
Why logic ensureUserNotSet doesn't work here?
I see a lot of doubling comparing with `setUser` can we somehow create some general method?

Reply:
`ensureUserNotSet` uses `InitConnectionListener` which is not used in `setUserWithoutConnecting` because we are not starting WS connection.

I'll try to extract common parts of code.